addons/purchase_stock/models/stock.py

```diff
@@ -163,10 +164,84 @@ def _prepare_move_default_values(self, return_line, new_picking):
 class Orderpoint(models.Model):
     _inherit = "stock.warehouse.orderpoint"
 
+    show_supplier = fields.Boolean('Show supplier column', compute='_compute_show_suppplier')
+    supplier_id = fields.Many2one(
+        'product.supplierinfo', string='Vendor',
+        domain="['|', ('product_id', '=', product_id), '&', ('product_id', '=', False), ('product_tmpl_id', '=', product_tmpl_id)]")
+
+    def _get_default_route_id(self):
+        if self.product_id._prepare_sellers():
+            route_id = self.env['stock.rule'].search([
+                ('action', '=', 'buy')
+            ]).route_id
+            if route_id:
+                return route_id[0]
+        return super()._get_default_route_id()
+
+    def _commpute_allowed_route_ids(self):
+        super()._commpute_allowed_route_ids()
+        if not self.user_has_groups('stock.group_adv_location'):
+            route_buy = self.env['stock.rule'].search([('action', '=', 'buy')]).route_id
+            for orderpoint in self:
+                orderpoint.allowed_route_ids |= route_buy
+
+    def _compute_show_suppplier(self):
+        for orderpoint in self:
+            orderpoint.show_supplier = any(ru.action == 'buy' for ru in orderpoint.route_id.rule_ids)
+
+    def _get_quantity_in_progress(self, product_ids, location_ids=False, warehouse_ids=False):
```

Comment:
_quantity_in_progress and _get_quantity_in_progress next to each other?

Reply:
maybe move this method on product.product and excute them on `product_ids` instead of passing them?